test/controllers/graphql_controller_v2_test.rb

```diff
@@ -252,6 +252,188 @@ def setup
+                                        value
+                                      }
+                                    }
+                                  }
+                                }' }
+      assert_equal(200, response.status)
+      json = JSON.parse(response.body)
+
+      assert_nil(json['errors'])
+      eastern_hits = json['data']['search']['hits']
+      assert(eastern_hits.positive?)
+    end
+    VCR.use_cassette('graphqlv2 geobox western hemisphere') do
+      post '/graphql', params: { query: '{
+                                  search(geobox: {
+                                    minLongitude: 180,
+                                    minLatitude: -90,
+                                    maxLongitude: 0,
+                                    maxLatitude: 90
+                                  }) {
+                                    hits
+                                    records {
+                                      title
+                                      locations {
+                                        geoshape
+                                        kind
+                                        value
+                                      }
+                                    }
+                                  }
+                                }' }
+      assert_equal(200, response.status)
+      json = JSON.parse(response.body)
+
+      assert_nil(json['errors'])
+      western_hits = json['data']['search']['hits']
+      assert(western_hits.positive?)
+    end
+    refute_equal(eastern_hits, western_hits)
+  end
+
+  test 'graphqlv2 geobox search with keyword search' do
+    VCR.use_cassette('graphqlv2 geobox with keyword') do
+      post '/graphql', params: { query: '{
+                                  search(
+                                    searchterm: "train stations",
+                                    geobox: {
+                                      minLongitude: -73.507,
+                                      minLatitude: 41.239,
+                                      maxLongitude: -69.928,
+                                      maxLatitude: 42.886
+                                    }
+                                  ) {
+                                    hits
+                                    records {
+                                      title
+                                      locations {
+                                        geoshape
+                                        kind
+                                        value
+                                      }
+                                    }
+                                  }
+                                }' }
+      assert_equal(200, response.status)
+      json = JSON.parse(response.body)
+
+      assert_nil(json['errors'])
+      assert(json['data']['search']['hits'].positive?)
+    end
+  end
+
+  test 'graphqlv2 geobox search with geodistance search' do
+    # Should this work? It does.
```

Comment:
I think a slightly better comment might point out that we don't intend this to be a feature, but we aren't going to prevent people from doing it maybe? I guess we could suggest in the docs that only one geo query at a time is supported.

From my exploration around this I think it ANDs the results together but our data is difficult to reason with at times as we have a lot of things that show up effectively globally that feel like they shouldn't so it's hard to know if our queries are wonky or the data is wonky. For sure I get a smaller set of results when I search for a geobox around Boston and combine with with a geodistance search around the center of India than either of those searches independently which feels correct (even if some of the results are questionable... like "Rivers of Alaska" I'd expect not to be in that search but it is)

Reply:
I've updated this code comment - will be merging now. Thanks!